worlds/osrs/__init__.py

```diff
@@ -0,0 +1,640 @@
+        region = self.region_name_to_data["Menu"]
+        if len(location_row.regions) > 0:
+            region = self.region_name_to_data[location_row.regions[0]]
+        location.parent_region = region
+        region.locations.append(location)
+
+    def set_rules(self) -> None:
+        """
+        called to set access and item rules on locations and entrances.
+        """
+        quest_attr_names = ["Cooks_Assistant", "Demon_Slayer", "Restless_Ghost", "Romeo_Juliet",
+                            "Sheep_Shearer", "Shield_of_Arrav", "Ernest_the_Chicken", "Vampyre_Slayer",
+                            "Imp_Catcher", "Prince_Ali_Rescue", "Dorics_Quest", "Black_Knights_Fortress",
+                            "Witchs_Potion", "Knights_Sword", "Goblin_Diplomacy", "Pirates_Treasure",
+                            "Rune_Mysteries", "Misthalin_Mystery", "Corsair_Curse", "X_Marks_the_Spot",
+                            "Below_Ice_Mountain"]
+        for qp_attr_name in quest_attr_names:
+            loc_name = getattr(LocationNames, f"QP_{qp_attr_name}")
+            item_name = getattr(ItemNames, f"QP_{qp_attr_name}")
+            self.multiworld.get_location(loc_name, self.player) \
+                .place_locked_item(self.create_event(item_name))
+
+        for quest_attr_name in quest_attr_names:
+            qp_loc_name = getattr(LocationNames, f"QP_{quest_attr_name}")
+            q_loc_name = getattr(LocationNames, f"Q_{quest_attr_name}")
+            add_rule(self.multiworld.get_location(qp_loc_name, self.player), lambda state, q_loc_name=q_loc_name: (
+                self.multiworld.get_location(q_loc_name, self.player).can_reach(state)
+            ))
+
+        # place "Victory" at "Dragon Slayer" and set collection as win condition
+        self.multiworld.get_location(LocationNames.Q_Dragon_Slayer, self.player) \
+            .place_locked_item(self.create_event("Victory"))
+        self.multiworld.completion_condition[self.player] = lambda state: (state.has("Victory", self.player))
+
+        for location_name, location in self.location_name_to_data.items():
+            location_row = self.location_rows_by_name[location_name]
+            # Set up requirements for region
+            for region_required_name in location_row.regions:
+                region_required = self.region_name_to_data[region_required_name]
+                add_rule(location, lambda state, region_required=region_required: state.can_reach(region_required, "Region",
+                                                                                                  self.player))
+            for skill_req in location_row.skills:
+                add_rule(location, self.get_skill_rule(skill_req.skill, skill_req.level))
+            for item_req in location_row.items:
+                add_rule(location, lambda state, item_req=item_req: state.has(item_req, self.player))
+            if location_row.qp != 0:
```

Comment:
Could do this instead
```suggestion
            if location_row.qp:
```